service-worker.js

```diff
@@ -0,0 +1,22 @@
+---
+---
+
+importScripts('/scripts/sw-toolbox.js');
+
+const VERSION = 1;
+
+toolbox.cache.name = "Babel-Cache-" + VERSION;
```

Comment:
👍  to Jeff's comment. You'll want to iterate through your cache entries and `cache.delete` doing something along these lines:

```js
// Where urlsToCacheKeys is a map of your items to cache
self.addEventListener('activate', function(event) {
  let setOfExpectedUrls = new Set(urlsToCacheKeys.values());
  event.waitUntil(
    caches.open(cacheName).then(function(cache) {
      return cache.keys().then(function(existingRequests) {
        return Promise.all(
          existingRequests.map(function(existingRequest) {
            if (!setOfExpectedUrls.has(existingRequest.url)) {
              return cache.delete(existingRequest);
            }
          })
        );
      });
    }).then(function() {
      return self.clients.claim();
    })
  );
});
```